src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/magical/talismans/ResurrectedTalisman.java

```diff
@@ -0,0 +1,95 @@
+package io.github.thebusybiscuit.slimefun4.implementation.items.magical.talismans;
+
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParser;
+import io.github.thebusybiscuit.slimefun4.core.handlers.ItemUseHandler;
+import io.github.thebusybiscuit.slimefun4.implementation.SlimefunPlugin;
+import me.mrCookieSlime.Slimefun.api.SlimefunItemStack;
+import org.bukkit.Bukkit;
+import org.bukkit.Location;
+import org.bukkit.NamespacedKey;
+import org.bukkit.World;
+import org.bukkit.entity.Item;
+import org.bukkit.inventory.ItemStack;
+import org.bukkit.persistence.PersistentDataContainer;
+import org.bukkit.persistence.PersistentDataType;
+import org.bukkit.potion.PotionEffect;
+import org.bukkit.potion.PotionEffectType;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import java.util.UUID;
+
+/**
+ * This {@link SlimefunItem} is a special variant of the {@Link Talisman}. This specific {@Link Talisman} saves you
+ * from a death caused by the void.
+ *
+ * @author svr333
+ *
+ */
+
+public class ResurrectedTalisman extends Talisman {
+    private static final NamespacedKey locationKey = new NamespacedKey(SlimefunPlugin.instance(), "resurrected_location");
+
+    public ResurrectedTalisman(SlimefunItemStack item, ItemStack[] recipe) {
+        super(item, recipe, true, true, "resurrected", new PotionEffect(PotionEffectType.GLOWING, 400, 0), new PotionEffect(PotionEffectType.ABSORPTION, 400, 4), new PotionEffect(PotionEffectType.DAMAGE_RESISTANCE, 200, 20));
+
+        addItemHandler(getItemHandler());
+    }
+
+    public ItemUseHandler getItemHandler() {
+        return e -> {
+            Location currentLoc = e.getPlayer().getLocation();
+            JsonObject json = createJsonFromLocation(currentLoc);
+
+            PersistentDataContainer pdc = e.getItem().getItemMeta().getPersistentDataContainer();
+            pdc.set(locationKey, PersistentDataType.STRING, json.toString());
```

Comment:
`.getItemMeta()` creates a copy of the original item meta, you will need to save the ItemMeta as a variable, modify it and then set the Itemmeta back to the ItemStack using `setItemMeta`, otherwise your changes won't be applied to the item.

Reply:
I found the itemMeta part really complicated overall, but understand it better now.
Is it okay if I add a new page to the developer guide on the wiki on how to do this?

Reply:
I mean sure, although I am not quite sure on how to generalize this but feel free to!